### DISCUSSION TITLE
Set task order but not dependency

### DISCUSSION THREAD

**Discussion by igonro** [PrefectHQ/prefect#5329] [Q&A] [answered]
I have 2 tasks that will be running in the same flow. I want to always execute first task_a, and then task_b, with the functional API I have the following code:
```python
with Flow("my-flow", schedule=every_5_min) as flow:
    param_a = Parameter("param_a", default=False)
    param_b = Parameter("param_b", default=0.8)
    param_c = Parameter("param_c", default="2021-01-01 00:00:00")
    task(my_func_a)(param_a)
    task(my_func_b)(param_a, param_b, param_c)
flow.register(project_name="my-project")
```
Sometimes task_b gets executed before task_a, I guess because they can run independently, but I prefer to run task_a before because it can modify some values that will use task_b (on a database). I tried using the Imperative API in order to make them dependent.

```python
flow = Flow("my-flow", schedule=every_5_min)
param_a = Parameter("param_a", default=False)
param_b = Parameter("param_b", default=0.8)
param_c = Parameter("param_c", default="2021-01-01 00:00:00")
task_a = task(my_func_a)
task_b = task(my_func_b)
flow.add_task(task_a)
flow.add_task(task_b)
task_a.set_downstream(task_b, flow=flow)
task_a.bind(param_a=param_a, flow=flow)
task_b.bind(param_a=param_a, param_b=param_b, param_c=param_c, flow=flow)
flow.register(project_name="my-project")
```

But this way, if task_a fails, task_b will not get executed.

Is it possible to make that task_b gets executed even when task_a fails, but forcing to execute task_a always before task_b?

Thanks!

**Comment by kvnkho** [comment, marked as answer]
You should just set_upstream in the functional API

```python
with Flow("my-flow", schedule=every_5_min) as flow:
    param_a = Parameter("param_a", default=False)
    param_b = Parameter("param_b", default=0.8)
    param_c = Parameter("param_c", default="2021-01-01 00:00:00")
    first = task(my_func_a)(param_a)
    second = task(my_func_b)(param_a, param_b, param_c)
    second.set_upstream(first)
flow.register(project_name="my-project")
```

And then if task_a fails, to get task_b to run, you want to look into [triggers](https://docs.prefect.io/api/latest/triggers.html) and set a trigger `always_run`

**Reply by igonro** [reply]
Hi @kvnkho, thanks for your help!
